sql/plan/empty_table.go

```diff
@@ -19,12 +19,24 @@ import "github.com/dolthub/go-mysql-server/sql"
 // EmptyTable is a node representing an empty table.
 var EmptyTable = new(emptyTable)
 
-type emptyTable struct{}
+func IsEmptyTable(n sql.Node) bool {
+	_, ok := n.(*emptyTable)
+	return ok
+}
+func NewEmptyTableWithSchema(schema sql.Schema) sql.Node {
+	return &emptyTable{schema: schema}
+}
+
+var _ sql.Node = (*emptyTable)(nil)
+
+type emptyTable struct {
+	schema sql.Schema
+}
 
-func (emptyTable) Schema() sql.Schema   { return nil }
-func (emptyTable) Children() []sql.Node { return nil }
-func (emptyTable) Resolved() bool       { return true }
-func (e *emptyTable) String() string    { return "EmptyTable" }
+func (e *emptyTable) Schema() sql.Schema { return e.schema }
+func (emptyTable) Children() []sql.Node  { return nil }
+func (emptyTable) Resolved() bool        { return true }
```

Comment:
might be paranoid, but having these as pointer receivers might be safer

Reply:
For the methods that do not use the receiver, it looks like we can change the receiver types to pointers:
```
func (*emptyTable) Children() []sql.Node { return nil }
```

Is that what you mean?

Reply:
yep

Reply:
Updated.